docs/design-documents/hal/0005-pin-names-Arduino-Uno-standard.md

```diff
@@ -0,0 +1,144 @@
+# Arduino Uno Pin Names design document
+
+# Table of contents
+
+1. [Table of contents](#table-of-contents).
+    * [Revision history](#revision-history).
+1. [Introduction](#introduction).
+    1. [Overview and background](#overview-and-background).
+    1. [Requirements and assumptions](#requirements-and-assumptions).
+1. [Detailed design](#detailed-design).
+    1. [Arduino Uno (Rev 3) connector Pins](#arduino-uno-rev-3-connector-pins).
+        1. [I2C and SPI Definition for Arduino Uno (Rev3) Pins](#i2c-definition-for-arduino-uno-rev3-pins)
+    1. [Board Components Pins](#board-components-pins).
+        1. [LED Definition](#led-definition)
+        1. [BUTTON Definition](#button-definition)
+1. [Other information](#other-information).
+    1. [Avoid Unnecessary Definitions](#avoid-unnecessary-definitions).
+
+
+### Revision history
+
+1.0 - Initial revision - Malavika Sajikumar, Marcelo Salazar - August 2020  
+This document is written for Mbed OS 6.
+
+# Introduction
+
+### Overview and background
+
+Mbed is designed so that application code written in the platform is portable across different Mbed supported boards with the same hardware capabilities or interfaces. However, the code, in most cases, is not truly portable due to the differences in pin name definitions for the same kind of interfaces across different boards. 
+
+This design document provides rules and guidelines on how to define Arduino Uno connector pins in the board support package of development boards to achieve true code portability across various boards with Arduino Uno connectors.
+
+### Requirements and assumptions
+
+This document applies to the pin standards required for Arduino Uno connector which is used on multiple Mbed Enabled boards. The Arduino Uno connector has been stable since 2012 at its current revision, which is the Arduino Uno Rev3. All design choices in this document for the Arduino Uno connector are based on the Arduino Uno Rev3 connector implementation.
+
+If the development board is defined as Arduino Uno compliant, the Arduino Uno connector standard has to be followed as described in this document and `ARDUINO_UNO` name should be defined as a supported form factor in targets.json file:
+
+    "supported_form_factors": [
+            "ARDUINO_UNO"
+        ],
+
+The Arduino Uno connector pins are defined in PinNames.h. The files resides in the following locations:
+
+    targets/MCU_VENDOR/MCU_FAMILY/MCU_NAME/Board/PinNames.h
+
+# Detailed design
+
+To achieve meaningful portability of application code across various Mbed Enabled boards that are Arduino Uno compliant, the pin names used for the connector pins should be common across these boards. This document describes a set of rules on how and where to define these pins in the board support package.
+
+### Arduino Uno (Rev3) connector Pins
+
+The following diagrams shows the Arduino Uno Rev3 standard for Mbed boards:
+
+![Static pinmap model](img/Arduino_Uno_Mbed.png)
```

Comment:
Need to describe timers. 
We should expand in text and explain recommendation and flag potential differences across board.
But we may not be in a position to reinforce compliance with Arduino Uno.